File: backend/internal/domain/station.go
package domain

import (
	"context"
	"time"
)

type Station struct {
	Id             int
	Ip             string
	ConnectedAt    time.Time
	DisconnectedAt *time.Time
}

type StationRepository interface {
	GetAll(ctx context.Context) ([]Station, error)
	Upsert(ctx context.Context, ip string) error
	UpdateDisconnectedAt(ctx context.Context, ip string) error
}

File: backend/src/domain/repositories/station.rs
use async_trait::async_trait;

use crate::domain::types::Station;
use crate::error::AppError;

#[async_trait]
pub trait StationRepository: Send + Sync {
    async fn get(&self, ip: &str) -> Result<Station, AppError>;
    async fn get_all(&self) -> Result<Vec<Station>, AppError>;
    async fn upsert(&self, ip: &str) -> Result<(), AppError>;
    async fn delete(&self, ip: &str) -> Result<(), AppError>;
}
